build.zig

```diff
@@ -782,7 +782,13 @@ fn addCmakeCfgOptionsToExe(
                 mod.linkSystemLibrary("unwind", .{});
             },
             .ios, .macos, .watchos, .tvos, .visionos => {
-                mod.link_libcpp = true;
+                if (static or !std.zig.system.darwin.isSdkInstalled(b.allocator)) {
+                    mod.link_libcpp = true;
+                } else {
+                    const sdk = std.zig.system.darwin.getSdk(b.allocator, b.graph.host.result) orelse return error.SdkDetectFailed;
+                    const @"libc++" = b.pathJoin(&.{ sdk, "usr/lib/libc++.tbd" });
+                    exe.root_module.addObjectFile(.{ .cwd_relative = @"libc++" });
```

Comment:
Note that you are implicitly assuming that the shared LLVM is linked against the system libc++. That's likely generally true in practice, but in theory one can build their own LLVM and link it against a non-system libc++.

I'm not suggesting that you need to support an LLVM built with a custom libc++. However, it may be worth documenting somewhere that you only support linking to a shared LLVM with the system libc++.

Reply:
Thank you; good advice. I think we will go the route of documenting requirement for only one copy of libc++. This problem will fully go away with #16270.